### PR TITLE
Visual Studio 2015 support

### PR DESCRIPTION
**Pull request by ximinez**
Reviewers: @miguelportilla @vinniefalco for their experience building in Windows. @rec may want to take a look, too, because I know he has SConstruct changes pending.

Installation documentation is now located in Builds/VisualStudio2015/README.md. Once these changes are merged / released, the [wiki page](https://wiki.ripple.com/Visual_Studio_2013_Build_Instructions) will need to be redirected to that file (possibly with a note that 2013 is no longer supported). When/if the 3rd party OpenSSL installer we rely on is updated to support VS2015, we can decided whether to update *SConstruct and the README* to reference it, or let users continue to build themselves. (Alternately, if the consensus is to maintain compatibility, I could revert the relevant SConstruct changes, and add instructions to move and rename files appropriately. I think this is a bad idea.)

### CODE REVIEW

**Comment by miguelportilla** [comment]
Builds all unity and non unity debug/release builds. All tests pass. Server starts, syncs and stops.

:+1:

**Comment by rec** [comment]
AppVeyor doesn't like it, though.  :-(

For those of us without access to a Windows machine, this is the only way of seeing if our builds work.  Is there a way to preserve AppVeyor through the move?

**Comment by ximinez** [comment]
@rec I noticed that's not a spurious failure either, so I'll try to fix it.

**Comment by ximinez** [comment]
YEAHEHEHEHEHEAAAAAAH!  "AppVeyor build succeeded".

**Comment by sublimator** [comment]
W0000000t

**Comment by vinniefalco** [comment]
Now THIS is good news!

**Comment by ximinez** [comment]
Relevant PR at https://github.com/ripple/Downloads/pull/2. If/when it is merged, we can change the `RIPPLED_DEPS_URL` in `appveyor.yml` to point to a more official Ripple repo.

**Comment by HowardHinnant** [comment]
Merge now, merge now!!!  Before ci/circleci whines! :-)

**Comment by sublimator** [comment]
lol :)

**Comment by ximinez** [comment]
Split the beast changes into a separate commit, and squashed the rest.

**Comment by ximinez** [comment]
Rebased to `b6`.

File: Builds/VisualStudio2015/README.md [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+
+## Change the Visual Studio Projects Debugging Properties
+
+1. If not already open, open the solution file located at **Builds/Visual Studio 2015/Ripple.sln**
+2. Select the correct solution platform in the solution platform dropdown (either *x64*
+  or *Win32* depending on machine type).
+3. Select the "Project->Properties" menu item to bring up RippleD's Properties Pages
+4. In "Configuration Properties" select "Debugging".
+5. In the upper-left Configurations drop down, select "All Configurations".
+6. In the upper-left Configurations drop down, select "All Configurations".
```
Comment (seelabs) [inline Builds/VisualStudio2015/README.md]: This line is a repeat of (5).

File: Builds/VisualStudio2015/README.md [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+git checkout master
+```
+
+* If you receive an error about not having the "correct access rights"
+  make sure you have Github ssh keys, as described above.
+
+### Configure Library Paths
+
+Open the solution file located at **Builds/Visual Studio 2015/ripple.sln**
+and select the "View->Property Manager" to bring up the Property Manager.
```
Comment (seelabs) [inline Builds/VisualStudio2015/README.md]: This is in `View->Other Windows->Property Manager` for me.

**Comment by vinniefalco** [comment]
This is already merged, so any changes would have to go into a new pull request.

File: Builds/VisualStudio2015/README.md [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+Expand the *debug | x64* section (or similar section on 32-windows) and
+double click the *Microsoft.Cpp.Win32.user* property sheet to bring up the
```
Comment (seelabs) [inline Builds/VisualStudio2015/README.md]: `Microsoft.Cpp.x64.user` (not `Win32`) in the expanded `x64` section

File: Builds/VisualStudio2015/README.md [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+4. In "Configuration Properties" select "Debugging".
+5. In the upper-left Configurations drop down, select "All Configurations".
+6. In the upper-left Configurations drop down, select "All Configurations".
+7. In "Debugger to Launch" select "Local Windows Debugger".
+
+### Tell rippled where to find the configuration file.
+
+The `--conf` command-line switch to tell rippled where to find this file.
+In the "Command Arguments" field in the properties dialog (that you opened
+in the above section), add: `--conf="C:\\Users\\joe\\ripple\\config\\rippled.cfg"`
```
Comment (seelabs) [inline Builds/VisualStudio2015/README.md]: Forward slashes should work. (i.e. `--conf="C:/Users/joe/ripple/config/rippled.cfg"`)

**Pull request closed without merging**